packages/node/src/exec/win32.ts

```diff
@@ -1,119 +1,110 @@
+
+      let join = (): Operation<ExitStatus> => function*() {
+        let result: Result = yield getResult.promise;
+        if (result.type === "status") {
+          let [code, signal] = result.value;
+          return { command, options, code, signal };
+        } else {
+          throw result.value;
+        }
+      }
 
-      task.spawn(
-        on<string>(childProcess.stderr, "data").forEach((data) => {
-          addToTail(data);
-          stderr.send(data);
-        })
-      );
+      let expect = (): Operation<ExitStatus> => function*() {
+        let status: ExitStatus = yield join();
+        if (status.code != 0) {
+          throw new ExecError(status, command, options);
+        } else {
+          return status;
+        }
+      }
 
-      task.spawn(
-        stdin.forEach((data) => {
+      let childProcess = spawnProcess(command, options.arguments || [], {
+        // We lose exit information and events if this is detached in windows
+        // and it opens a window in windows+powershell.
+        detached: false,
+        // When windows shell is true, it runs with cmd.exe by default, but
+        // node has trouble with PATHEXT and exe. It can't run exe directly for example.
+        // `cross-spawn` handles running it with the shell in windows if needed.
+        // Neither mac nor linux need shell and we run it detached.
+        shell: false,
+        // With stdio as pipe, windows gets stuck where neither the child nor the
+        // parent wants to close the stream, so we call it ourselves in the exit event.
+        stdio: "pipe",
+        // Hide the child window so that killing it will not block the parent
+        // with a Terminate Batch Process (Y/n)
+        windowsHide: true,
+
+        env: options.env,
+        cwd: options.cwd,
+      });
+
+      let { pid } = childProcess;
+
+      let stdoutChannel = createChannel<string>();
+      let stderrChannel = createChannel<string>();
+      let stdin = {
+        send(data: string) {
           childProcess.stdin.write(data);
-        })
-      );
-
-      let value = yield onceEmit(childProcess, "exit");
-      getResult.resolve({ type: "status", value });
-    } finally {
-      stdout.close();
-      stderr.close();
-      if (pid) {
-        ctrlc(pid);
-        let stdin = childProcess.stdin;
-        if (stdin.writable) {
-          try {
-            //Terminate batch process (Y/N)
-            stdin.write("Y\n");
-          } catch (_err) { /* not much we can do here */}
         }
-        stdin.end();
+      };
+
+      scope.spawn(function*(task) {
+        task.spawn(function*() {
+          let value: Error = yield once(childProcess, 'error');
+          getResult.resolve({ type: 'error', value });
+        });
+
+        task.spawn(on<Buffer>(childProcess.stdout, 'data').map((c) => c.toString()).forEach(stdoutChannel.send));
+        task.spawn(on<Buffer>(childProcess.stderr, 'data').map((c) => c.toString()).forEach(stderrChannel.send));
```

Comment:
It's nice that we have bound functions now for all instances so that we can use `channel.send` as a value.